src/czar/Czar.cc

```diff
@@ -538,6 +539,86 @@ void Czar::removeOldResultTables() {
+    }
+    vector<string> colStatus;
+    vector<string> colMessageTable;
+    vector<string> colResultQuery;
+    if (!results.extractFirst3Columns(colStatus, colMessageTable, colResultQuery, err)) {
+        string const msg = context + "Failed to extract info for the user query, err=" + err.printErrMsg() +
+                           ", sql=" + sql;
+        throw runtime_error(msg);
+    }
+    if (colStatus.size() != 1) {
+        string const msg = context + "Unknown user query, err=" + err.printErrMsg() + ", sql=" + sql;
+        throw runtime_error(msg);
+    }
+
+    SubmitResult result;
+    if (colStatus[0] == "FAILED") {
+        result.errorMessage = "The query failed";
+    } else if (colStatus[0] == "FAILED") {
+        result.errorMessage = "The query was aborted";
+    }
+    result.resultTable = "result_" + to_string(queryId);
+    result.messageTable = colMessageTable[0];
+    result.resultQuery = colResultQuery[0];
+    result.queryId = queryId;
+    result.status = colStatus[0];
+
+    // Pull ongoing query processing stats if this information is still available.
+    // This is a transient information located in the temporary table.
+    // It's available for the duration of the query processing.
+    sql = "SELECT totalChunks,completedChunks,UNIX_TIMESTAMP(queryBegin),UNIX_TIMESTAMP(lastUpdate) FROM "
+          "QStatsTmp WHERE queryId=" +
+          to_string(queryId);
+    if (!sqlConn->runQuery(sql, results, err)) {
+        string const msg = context +
+                           "Query to find stats for the user query failed, err=" + err.printErrMsg() +
+                           ", sql=" + sql;
+        throw runtime_error(msg);
+    }
+    vector<string> colTotalChunks;
+    vector<string> colCompletedChunks;
+    vector<string> colQueryBeginEpoch;
+    vector<string> colLastUpdateEpoch;
+    if (!results.extractFirst4Columns(colTotalChunks, colCompletedChunks, colQueryBeginEpoch,
+                                      colLastUpdateEpoch, err)) {
+        string const msg = context + "Failed to extract stats for the user query, err=" + err.printErrMsg() +
+                           ", sql=" + sql;
+        throw runtime_error(msg);
+    }
+    switch (colTotalChunks.size()) {
+        case 0:
+            // No stats means the query is over
+            break;
+        case 1:
+            // The query might be still in progress
+            result.totalChunks = stoi(colTotalChunks[0]);
+            result.completedChunks = stoi(colCompletedChunks[0]);
+            result.queryBeginEpoch = stoi(colQueryBeginEpoch[0]);
+            result.lastUpdateEpoch = stoi(colLastUpdateEpoch[0]);
+            break;
+        default:
+            // Should never be here.
+            string const msg = context +
+                               "Inconsistent stats returned for the user query, err=" + err.printErrMsg() +
+                               ", sql=" + sql;
+            throw runtime_error(msg);
```

Comment:
This is not something to be answered in this ticket, but maybe we should make a QservRuntimeException class based off utill::Issue and use that instead of runtime_error. That makes it easy to catch qserv related errors and also includes the source code location of the throw if it is caught.

Reply:
Good idea! At some point, we would need to review error reporting at Czar as many of those can be reported to Qserv users. Some messages may be quite confusing to users.